### DISCUSSION TITLE
Is there some way to get the last insert id after an insert operation?

### DISCUSSION THREAD

**Discussion by importcjj** [weiznich/diesel_async#126] [Q&A]

**Comment by weiznich** [comment]
It's not possible to answer this without knowing the relevant context. Please at at least the following details to your questions:

* Information about your diesel-async and diesel version
* Information about your database system (which one is used + what version)
* Relevant parts of your schema, code, …

Additionally: The issue tracker is no support forum and the issue template is mandatory. I've moved your question to the right place this time. Try to place it correctly next time.

**Reply by importcjj** [reply]
diesel = { version = "2.1.0", features = ["mysql_backend", "time"] }
diesel-async = { version = "0.4.1", features = ["mysql", "deadpool"] }

I'm using Mysql as the backend, and  the schema is like:

```rust
diesel::table! {
    data_sets {
        id -> Unsigned<BigInt>,
        symbol -> Text,
        title -> Text,
        description -> Text,
        data_source -> Text,
        created_at -> Timestamp,
        updated_at -> Timestamp,
    }
}

#[derive(Debug, Serialize, Deserialize, Queryable, Selectable)]
pub struct DataSet {
    pub id:          u64,
    pub symbol:      String,
    pub title:       String,
    pub description: String,
    pub data_source: String,
    pub created_at:  time::OffsetDateTime,
    pub updated_at:  time::OffsetDateTime,
}

```

I want to get the  ID of my inserted record, but it only returns `affected_rows` now. Is there a way to get the ID?

```rust
        let r = diesel::insert_into(data_sets::table)
            .values((
                data_sets::symbol.eq(&params.symbol),
                data_sets::data_source.eq(&s),
                data_sets::title.eq(&params.title),
                data_sets::description.eq(&params.description),
            ))
            .execute(&mut conn)
            .await?;


```

**Reply by weiznich** [reply]
This seems to be an duplicate of https://github.com/diesel-rs/diesel/issues/1011 then. Solutions are listed in the linked issue.

**Reply by importcjj** [reply]
Ok, thanks for your reply
